enhancements/validate-catalog-for-bad-bundles.md

```diff
@@ -0,0 +1,177 @@
+    - [VerticalPodAutoScaler](https://cloud.google.com/kubernetes-engine/docs/concepts/verticalpodautoscaler)
+- Design a tool that lists all "bad" bundles for a given [catalog source](https://olm.operatorframework.io/docs/concepts/crds/catalogsource/)
+
+
+### Non-Goals
+
+- Server-side validation of Kube manifests as part of a controller
+
+## Open Questions
+    
+1. Should the tool provide an option to choose a container tool (`--image-builder` flag)?
+2. Potential changes to Scorecard?
+
+## Static Validation Rules
+
+### PDB yaml validation
+Example PDB.yaml
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: zk-pdb
+spec:
+  minAvailable: 2  
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+- Either `minAvailable` or `maxUnavailable` should be included, but not both
+- `minAvailable` field cannot be set to 0 or less
+- `maxUnavailable` cannot be set to greater than the number of pods (TODO: check with Dan)
+
+
+### PriorityClass yaml validation
+Example PriorityClass.yaml
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+- `globalDefault` should always be set to false
+
+
+### VerticalPodAutoScaler yaml validation
+Example VerticalPodAutoScaler.yaml
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: my-rec-vpa
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind:       Deployment
+    name:       my-rec-deployment
+  updatePolicy:
+    updateMode: "Off"
+```
+- 
+
+## Proposal
+
+The design proposal has two parts:
+- The first discusses potential approaches for validating SAP Gardener objects.
+- The second part discusses design approach for having a tool to list "bad" bundles in a catalog
+
+## Part 1: Validate bundle objects
+
+### 1. Extending Validation library to add validations in `Golang`
+- Extend [BundleValidator](https://github.com/operator-framework/api/blob/master/pkg/validation/internal/bundle.go#L14) to include new validation functions for each of the SAP Gardener objects. (ToDo: Do we add any checks under [BundleFormat](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/validate.go#L65) ?)
+- The validation function for each kube manifest object would check for conditions specified under [Static Validation Rules](https://hackmd.io/7TlDq9t_RuGjGm8YsEbyHA?both#Static-Validation-Rules).
+- (ToDo: 
+    -    Do we want to maintain some sort of `struct` for storing various kube-manifest objects as these objects could grow (in the future) or
+    -    Wait to see how these new objects are represented in the [Bundle](https://github.com/operator-framework/api/blob/master/pkg/manifests/bundle.go#L11) as part SAP gardener PR and then take a call)
```

Comment:
Should we be limiting the scope of this enhancement to a specific implementation (i.e. the SAP operator)? It seems like some of these concepts would be just generally useful, and this is just an example of the kind of objects would be allowed. In the future, if OLM was to support arbitrary manifests, it may be really useful to have a way to apply rulesets to validation here.

Reply:
Your comment makes sense to me but I am still unclear if it is inclined towards the 1st or the 2nd point I made in my ToDo